tools/psa/build_tfm.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/python
+"""
+Copyright (c) 2019 ARM Limited. All rights reserved.
+
+SPDX-License-Identifier: Apache-2.0
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import os
+from os.path import join, abspath, dirname, isdir
+import sys
+import subprocess
+import logging
+import argparse
+
+logger = logging.getLogger('TF-M-Builder')
+logging.basicConfig(level=logging.INFO,
+                    format='[%(name)s] %(asctime)s: %(message)s.',
+                    datefmt='%H:%M:%S')
+
+ROOT = abspath(join(dirname(__file__),
+                                    os.pardir, os.pardir))
+sys.path.insert(0, ROOT)
+from tools.targets import Target, TARGET_MAP, TARGET_NAMES
+
+TF_M_BUILD_DIR = join(ROOT, os.pardir, 'tfm_build_dir')
```

Comment:
Wouldn't it be good to use the `mbed-os/BUILD` directory to have everything there, as it is a dedicated build folder?

Reply:
We are planning to use the mbed-os importer to bring in TF-M PSA APIs for non-secure side. Therefore we cannot use mbed-os/BUILD directory. The importer script expects the source to be outside of mbed-os.
Another reason is not be interfere with mbed-cli building non-secure side (mbed-os). If we use mbed-os/BUILD then mbed-cli will try to compie tf-m sources when non-secure side is being built.

Reply:
Well, this would imply users can only have their changes saved locally, but can't pull in their git branches as the tag/version would change to a different HEAD. Is my understanding right?